packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -254,16 +259,27 @@ - (void)updateOrientation:(UIDeviceOrientation)orientation
 
 - (void)captureToFile:(FLTThreadSafeFlutterResult *)result {
   AVCapturePhotoSettings *settings = [AVCapturePhotoSettings photoSettings];
+  NSString *extension = @"jpg";
   if (_resolutionPreset == FLTResolutionPresetMax) {
     [settings setHighResolutionPhotoEnabled:YES];
   }
 
+  if (_outputFormat == OutputFormatHEIF) {
+    if (@available(iOS 11.0, *)) {
+      if ([self.capturePhotoOutput.availablePhotoCodecTypes containsObject:AVVideoCodecTypeHEVC]) {
```

Comment:
do you know when will this return false? what should we do if it's false? if we default to jpg, we probably wanna document that behavior too.

Reply:
https://github.com/flutter/packages/pull/4586#discussion_r1364741704